extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/ProcessCompilationProvider.java

```diff
@@ -23,6 +23,7 @@
     protected Generator addGenerator(ApplicationGenerator appGen, Set<File> filesToCompile, Context context)
             throws IOException {
         return appGen.withGenerator(
-                ProcessCodegen.ofFiles(new ArrayList<>(filesToCompile)));
+                ProcessCodegen.ofFiles(new ArrayList<>(filesToCompile)))
+                .withClassLoader(Thread.currentThread().getContextClassLoader());
```

Comment:
@mswiderski 

>                     .withPersistence(usePersistence) 

this has always been missing from here... not sure if it is a problem since it's for hot-reload only. Not sure how to fix it though (where to pull that boolean from while doing hot reload)